docs/reference/index-modules.asciidoc

```diff
@@ -46,6 +46,17 @@ NOTE: The number of shards are limited to `1024` per index. This limitation is a
 
 // end::index-number-of-shards-tag[]
 
+[[index-number-of-routing-shards]]
+`index.number_of_routing_shards`::
+Number of routing shards used to <<indices-split-index,split>> an index and its
+primary shards. This setting specifies the hashing space used internally to
+distribute documents for a split index across shards. See
+<<indices-split-index>>.
```

Comment:
I think this is too abstract. Shall we include the example from the split-index documentation here as well? It'd be a bit of duplication but I don't know if this is clear enough as to how it works without an example